src/syrupy/serializers/yaml.py

```diff
@@ -68,3 +81,31 @@ def _read_file(self, filepath: str) -> Any:
         except FileNotFoundError:
             pass
         return {}
+
+    def _read_raw_file(self, filepath: str) -> Dict[str, Optional[str]]:
+        """
+        Read the raw snapshot data (str) from the snapshot file into a dict
+        of snapshot name to raw data. This does not attempt any deserialization
+        of the snapshot data.
+        """
+        raw_snapshots = {}
+        cur_name = None
+        cur_data = None
+        try:
+            with open(filepath, "r") as f:
+                for line in f:
+                    indent = len(line) - len(line.lstrip(" "))
+                    if not indent:
+                        if cur_name is not None:
+                            raw_snapshots[cur_name] = cur_data
+
+                        cur_name = line[:-2]  # newline & colon
+                        cur_data = ""
+                    else:
+                        cur_data = f"{cur_data}{line[2:]}"
+                if cur_name is not None:
+                    raw_snapshots[cur_name] = cur_data
```

Comment:
I think this can be simplified by not having the `cur_data` staging
```py
        raw_snapshots = {}
        cur_name = None
        try:
            with open(filepath, "r") as f:
                for line in f:
                    indent = len(line) - len(line.lstrip(" "))
                    if not indent:
                        cur_name = line[:-2]  # newline & colon
                        raw_snapshots[cur_name] = ""
                    else:
                        raw_snapshots[cur_name] += line[2:]
```